src/coreComponents/dataRepository/LogLevelsRegistry.hpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file LogLevelsRegistry.hpp
+ */
+
+#ifndef GEOS_COMMON_LOGLEVELSREGISTRY_HPP
+#define GEOS_COMMON_LOGLEVELSREGISTRY_HPP
+
+#include "common/LogLevelsInfo.hpp"
+
+namespace geos
+{
+
+/**
+ * @brief Keep track of log level documention for a group
+ */
+class LogLevelsRegistry
+{
+public:
+
+  /**
+   * @brief Add a log description for a wrapper
+   * @param level The minimum log level
+   * @param description The description for the log level
+   */
+  void addEntry( integer level, std::string_view description );
+
+  /**
+   * @brief Construct the log level string description for a wrapper
+   * @return The log level string description
+   */
+  string buildLogLevelDescription();
+
+private:
+
+
+  /// Map for building the log level string for each wrapper
+  /// key : a logLevel condition, values : a set of description for a corresponding loglevel
+  std::map< integer, std::vector< std::string > > m_logLevelsDescriptions;
+
+};
+
+/**
+ * @brief Verify if a log level is active
+ * @tparam LOG_LEVEL_INFO The structure containing log level information.
+ * @param level Log level to be checked.
+ * @return `true` if the log level is active, `false` otherwise.
+ * @pre `LOG_LEVEL_INFO` must satisfy `logInfo::is_log_level_info`.
+ *
+ */
+template< typename LOG_LEVEL_INFO >
+std::enable_if_t< logInfo::is_log_level_info< LOG_LEVEL_INFO >, bool >
+isLogLevelActive( int level )
```

Comment:
Instead of directly giving the log level as parameter, would it be possible to have a templated type parameter `LOGGING_OBJECT const & loggingObject` for which we call a `loggingObject.getLogLevel()` method?
This would allow to ask if it has registered the required `LogLevelInfo`, and `GEOS_ERROR()` if it's not the case.

Reply:
- 1st  point : ReactiveCompositionalMultiphaseOBL : i didn't change this file
2st point : TODO

Reply:
As an exemple, I would expect a crash in this scenario:
```c++

```

another solution :
- require a `LogLevelRegistry const &` parameter in `isLogLevelActive()`
- modify the macros: `#define GEOS_LOG_LEVEL_INFO( logInfo, msg ) GEOS_INFO_IF( isLogLevelActive< logInfo >( this->getLogLevel(), this->getLogRegistry() ), msg );`
- reverse the include way for `LogLevelsRegistry.hpp` & `LogLevelsInfo.hpp`

also, please verify your headers, `Format.hpp` may be for cpps only.